packages/schema/src/plugins/zod/transformer.ts

```diff
@@ -114,7 +114,7 @@ export default class Transformer {
             } else if (inputType.type === 'Boolean') {
                 result.push(this.wrapWithZodValidators('z.boolean()', field, inputType));
             } else if (inputType.type === 'DateTime') {
-                result.push(this.wrapWithZodValidators('z.date()', field, inputType));
+                result.push(this.wrapWithZodValidators('z.union([z.date(), z.string().datetime()])', field, inputType));
```

Comment:
Hey, thanks for working on this!

I tried it out, and there's a problem with the distribution of `union` over `array`:

<img width="1004" alt="image" src="https://user-images.githubusercontent.com/104139426/232864442-5a8b1151-6ff8-418e-8685-a3c4b4176a7d.png">

<img width="802" alt="image" src="https://user-images.githubusercontent.com/104139426/232864648-cc43d816-9aaa-43b1-ae52-90c695f2664f.png">

Here the inferred type doesn't conform to Prisma filter's typing. The `union` needs to wrap around `array`. Maybe we should refactor `wrapWithZodValidators` and and pass in a list of validators so we can deal with the array distribution inside of it?

Reply:
Maybe, we have to do like if its a union, then use '.array()' for each of the rule inside union
Like if it's `z.union([z.date(), z.string().datetime()])`, it should do `z.union([z.date().array(), z.string().datetime().array()])`

Reply:
Yes, I think so.